paastech-cloud/git-repo-manager
# Refactor/repo-manager-as-root

## What does this PR do ?

This pull request aims to refactor the git-controller project by making the following change:

- Renaming the project from "git-controller" to "git-repo-manager" throughout the entire directory structure, starting from the root folder.

This pull request primarily aims to move the authentication layer and the file system virtualization layer (SSH key FUSE) from this repository to another repository called "git-ssh-server."

By separating these specific layers into the "git-ssh-server" repository, it allows for better modularization and organization of the codebase. This approach simplifies maintenance and future development by creating a dedicated repository focused on handling SSH authentication and handling database requests..

### How should this be manually tested ?

>**_Notes_**: Make sure to have your `.env` file set as needed it

To run the server, run the following command:

```bash
cargo run --bin repo-manager
```

To run the client-debug, run the following command:

```bash
cargo run --bin client-debug
```

### Other changes

- Modifying the `.toml` file to add two new targets: "client-debug" and "repo-manager". This change ensures that these targets are properly included and can be utilized in the project.

- Updating the documentation to reflect the refactored project structure.
